src/Url.php

```diff
@@ -1066,4 +1066,35 @@ public static function peelUrl($haystack, $prelude_string, $wrapper_start, $wrap
     }
     return FALSE;
   }
+
+  /**
+   * Checks if given URL matches a list of candidates for a default document.
+   *
+   * @param string $url
+   *   The URL to be tested
+   * @param array $candidates
+   *   A list of potential document names that could be indexes.
+   *   Defaults to "default" and "index".
+   *
+   * @return mixed
+   *   string - The base path if a matching document is found.
+   *   bool - FALSE if no matching document is found.
+   */
+  public static function getRedirectIfIndex($url, $candidates = array("default", "index")) {
+    // Assume no match
+    $root_path = FALSE;
+
+    // Filter through parse_url to separate out querystrings and etc.
+    $parsedUrl = parse_url($url);
+
+    // Test parsed URL
+    foreach ($candidates as $candidate) {
```

Comment:
Might be faster instead of foreaching the candidates, just to see if the
$filename = strtolower($pathinfo($parsedUrl["path"]));
if (!empt($filename) && in_array($filename, $candidates)) {